### DISCUSSION TITLE
Showing Unable to verify first certificate.

### DISCUSSION THREAD

**Discussion by sohagRudra** [nginx/kubernetes-ingress#6534] [Q&A]
when user do any https request(destination port-443) I am converting it to port 81 as http resuest.From there I am doing some processing and after that I am converting the destination port to 443 again.So basically I am trying to decode https into http and again encoding it to https.But when I am sending any https request through postman First the request is getting converted to http as I am getting some log in my processing.But then I think some issue happening while converting the http to https as I am getting 502(Bad Gateway) response and It is showing unable to verify first certificate.I have the same certificate and key while decoding and encoding.Can any one please give some insights on this?It is very urgent for me to figure out.Thanks.

**Comment by vepatel** [comment]
@sohagRudra can you please give us more details on what your configuration looks like i.e. ingress, svc and deployment?

**Reply by sohagRudra** [reply]
Hi  @vepatel ,
Actually My use case is like any https request generated from my laptop should be converted to http at port 81 where I will do ssl termination and then I will do some internal processing and after that I want to make it at same https request and hit the internet.I have done some progress.Till now I am being able to terminate the https to http(I am using self signed certificate) at port 81.I am doing processing there .But when I am redirecting it to https by using the server as 
server{
 listen 80 default_server;
 listen [::]:80 default_server;
 server_name _;
 return 301 https://$host$request_uri;
}
showing this -
cbnits-192@cbnits192-HP-250-G8-Notebook-PC:~$ curl --location 'https://api.escuelajs.co/api/v1/files/upload' --form 'file=@"/home/cbnits-192/Pictures/Screenshots/Screenshot from 2024-09-26 13-27-48.png"'
curl: (60) SSL certificate problem: self-signed certificate
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.

can you give some insights about it?

**Reply by j1m-ryan** [reply]
What version of NGINX Ingress Controller are you using? @sohagRudra

**Reply by sohagRudra** [reply]
nginx version: nginx/1.18.0 
I am using this version of nginx in my local set up.
@j1m-ryan @vepatel

**Reply by sohagRudra** [reply]
Actually I am doing https request from my terminal.Trying to use nginx to decrypt the https into http.Then again encrypting that http request to https.I am using self singed certificate generated using openssl in my nginx.
@j1m-ryan @vepatel

**Reply by j1m-ryan** [reply]
Hi @sohagRudra, this is quite an old NGINX Version, what version of NGINX Ingress Controller are you using? Can you describe the deployment?

**Reply by sohagRudra** [reply]
@j1m-ryan  .I think I am not using any ingress controller.I am just trying to control and modify my own request to a server.The server can be youtube or aws.I am not a devops guy so I may not get all your point .please pardon me.

**Comment by vepatel** [comment]
Hi @sohagRudra, since you're not using Nginx Ingress Controller, may I suggest you open a discussion in official Nginx github repo please.
https://github.com/nginx/nginx/discussions
